WcaOnRails/spec/models/user_spec.rb

```diff
@@ -173,11 +173,19 @@
       expect(user.wca_id).to be_nil
     end
 
-    it "verifies WCA ID unique when changing WCA ID" do
-      person2 = FactoryBot.create :person, wca_id: "2006FLEI01"
-      user2 = FactoryBot.create :user, wca_id: "2006FLEI01", name: person2.name
-      user.wca_id = user2.wca_id
-      expect(user).to be_invalid_with_errors(wca_id: ["must be unique"])
+    context "when WCA ID is not unique" do
+      let(:existing_user) { FactoryBot.create :user_with_wca_id }
+      let(:invalid_user) { FactoryBot.build :user, wca_id: existing_user.wca_id }
+
+      it "verifies WCA ID unique when changing WCA ID" do
+        expect(invalid_user.valid?).to be false
+        expect(invalid_user.errors.messages).to include :wca_id
+      end
+
+      it "shows an appropriate error message" do
+        expected_message = %(is already used by "#{existing_user.name}" with #{existing_user.email})
```

Comment:
Maybe just retrieve the message through i18n, similar to what you did in the other test:

```
I18n.t('users.errors.unique', used_name: existing_user.name, used_email: existing_user.email)
```
?

Reply:
I actually have a slight preference for the explicit string (it lets us known for sure that we got the string interpolation right, for instance), but I get that the duplication feels silly. @Istanful, I'd say just do whichever way you prefer!

Reply:
I say we keep it as is. Just as jfly said it actually serves a purpose. Still get why your gut is telling you otherwise though. In feature specs I would usually do it the other way, since in them we are actually testing that it is showing up rather than what it is showing.